sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MutableFieldsShouldNotBe.cs

```diff
@@ -101,6 +112,22 @@ private bool IsInvalidMutableType(ISymbol symbol)
             return false;
         }
 
+        private static bool IsValidReadOnlyInitializer(ISymbol symbol)
+        {
+            if (symbol == null)
+            {
+                return false;
+            }
+
+            var methodSymbol = symbol as IMethodSymbol;
+            if (methodSymbol != null)
+            {
+                return IsOrDerivesOrImplementsAny(methodSymbol.ReturnType, AllowedTypes);
+            }
+
+            return false;
+        }
+
         private static bool IsOrDerivesOrImplementsAny(ITypeSymbol typeSymbol, ISet<KnownType> knownTypes)

```

Comment:
Yeah, the naming is misleading. Frequently I am thinking about changing the names of these helpers a bit:
- Is --> IsEqual
- DerivesOrImplements --> Is

This way the semantic will be as in C#... WDYT?